src/init_interp/initInterp.F90

```diff
@@ -942,35 +943,55 @@ subroutine set_subgrid_info(beg, end, dimname, use_glob, ncid, active, subgrid)
 
   contains
 
-    subroutine read_var_double(ncid, varname, data, dim1name, use_glob)
+    subroutine read_var_double(ncid, varname, data, dim1name, use_glob, allow_scm)
       ! Wraps the ncd_io call, providing logic related to whether we're using the 'glob'
       ! form of ncd_io
       type(file_desc_t)  , intent(inout) :: ncid
       character(len=*)   , intent(in)    :: varname
       real(r8), pointer  , intent(inout) :: data(:)
       character(len=*)   , intent(in)    :: dim1name
       logical            , intent(in)    :: use_glob  ! if .true., use the 'glob' form of ncd_io
+      logical            , intent(in)    :: allow_scm ! if .true., use the 'glob' form of ncd_io
+
+      ! local
+      character(16)                      :: readflag
```

Comment:
We should add some comments here to explain what's going on with this scm/noscm stuff.

Reply:
Looking through this a bit, the comment right after `allow_scm` is an accidental copy from line 953 (directly above it). As such, I will replace the accidental comment with the one that I see elsewhere:
`      logical            , intent(in)    :: allow_scm ! if .true., allow scm data slice`

Reply:
I found an even better instance of the comment explaining `allow_scm`. Now using that one.

Reply:
Thanks Sam.  The comment should be changed as you recommend.  As for the read_noscm flag this is to distinguish those times during IC interpolation when you are running in single column mode but need to read the full data grid.  Normally single_column means read the data grid and extract the closest column but during IC interpolation you need to read in the full grid to be interpolated regardless of the single_column flag.